src/tests/memory_fs.rs

```diff
@@ -72,8 +72,8 @@ impl FileSystem for MemoryFS {
         self.metadata(path)
     }
 
-    fn read_link(&self, _path: &Path) -> io::Result<PathBuf> {
-        Err(io::Error::new(io::ErrorKind::NotFound, "not a symlink"))
+    fn read_link(&self, _path: &Path) -> Result<PathBuf, ResolveError> {
+        Err(io::Error::new(io::ErrorKind::NotFound, "not a symlink").into())
```

Comment:
```suggestion
        Err(Error::new(ErrorKind::NotFound, "not a symlink").into())
```

Let's be consistent.

Reply:
Which `Error` did you mean? L76 means

```rust
Err(ResolveError::from(io::Error::new(io::ErrorKind::NotFound, "not a symlink")))
```

Simply removing `io::` prefix will lead to error.